gpu_direct_rdma_access.c

```diff
@@ -448,6 +453,50 @@ static int modify_source_qp_to_rtr_and_rts(struct rdma_device *rdma_dev)
     return 0;
 }
 
+static int destroy_qp(struct ibv_qp *qp) 
+{
+    int ret;
+    if (qp) {
+	ret = ibv_destroy_qp(qp);
+        DEBUG_LOG("ibv_destroy_qp(%p): %d\n", qp, ret);
```

Comment:
based on another comment here.... lets move the "fprintf(stderr,...) " as the error logger here.

and then move the DEBUG_LOG back to top of function (before teh ibv_destroy_qp() call) as you did in the first place.